src/Mantasflowers.Services/Services/Coupon/CouponService.cs

```diff
@@ -25,6 +25,7 @@ public async Task<Domain.Entities.Coupon> CreateCouponAsync(PostCreateCouponRequ
             try
             {
                 coupon = await _unitOfWork.CouponRepository.CreateAsync(coupon);
+                await _unitOfWork.SaveChangesAsync();
```

Comment:
Both SaveChangesAsync() are called in PaymentService, since CouponService and OrderService can't function by themselves. In a way, it's a kind of prevention from misuse.